### PR TITLE
add: integrationType to idportenclient

### PR DESCRIPTION
**Pull request by ybelMekk**

### CODE REVIEW

**Review by tronghn — COMMENTED** [review]
Things to consider:

- Do we also need to open for the configuration of the `scopes` field for the IDPortenClient? We only register two scopes as of today: `[]string{"openid", "profile"}`.
- Remove the extra step for deletion of clients: https://doc.nais.io/security/auth/idporten/#permanently-deleting-a-client. As it stands, this will likely cause some confusion when users inevitably attempt to change their integration type.

**Comment by ybelMekk** [comment]
True true..

**Review by tronghn — APPROVED** [review]
LGTM otherwise. Any changes for the considerations above could probably be introduced in separate pull requests.

**Comment by ybelMekk** [comment]
hmm, i think i add it to the same PR. Its not to many lines to add.

File: pkg/clients/instance.go [hunk 1]
```diff
@@ -146,9 +149,27 @@ func toIDPortenClientRegistration(in naisiov1.IDPortenClient) types.ClientRegist
 	if in.Spec.PostLogoutRedirectURIs == nil || len(in.Spec.PostLogoutRedirectURIs) == 0 {
 		in.Spec.PostLogoutRedirectURIs = []naisiov1.IDPortenURI{IDPortenDefaultPostLogoutRedirectURI}
 	}
+	if in.Spec.IntegrationType != "" {
+		switch in.Spec.IntegrationType {
+		case string(types.IntegrationTypeIDPorten):
+			if in.Spec.Scopes == nil {
```
Comment (tronghn) [inline pkg/clients/instance.go:155]: `in.Spec.Scopes` is never nil. Should instead check for emptyiness here:

```suggestion
			if len(in.Spec.Scopes) == 0 {
```

File: pkg/clients/instance.go [hunk 2]
```diff
@@ -146,9 +149,27 @@ func toIDPortenClientRegistration(in naisiov1.IDPortenClient) types.ClientRegist
...
+				in.Spec.Scopes = defaultValidIdportenScopes
+			}
+		case string(types.IntegrationTypeKrr):
+			if in.Spec.Scopes == nil {
```
Comment (tronghn) [inline pkg/clients/instance.go:159]:
```suggestion
			if len(in.Spec.Scopes) == 0 {
```

**Review by tronghn — APPROVED** [review]

**Pull request merged**